Support Debian (.deb) packaging

cxw42 opened the pull request:
Thanks for this plugin, which is very useful!  At D3 we use .deb most often.  This PR adds the structure to build .deb files directly from this Git repo (rather than from a dist tarball).  I have tested this on Jetson (aarch64) and PC (amd64).  Thank you for considering this PR!

(If you get to this in October, would you be willing to label it `hacktoberfest-accepted`?  [Hacktoberfest is opt-in](https://hacktoberfest.digitalocean.com/hacktoberfest-update) this year. :D  I appreciate your consideration!)

Commit:
Add LICENSE file [minor]

Copied from <https://www.gnu.org/licenses/old-licenses/lgpl-2.0.txt>.

Also, update README to list license.

michaelgruner commented on README.md:
```diff
@@ -24,3 +24,18 @@ encoder:
+
+## Legal
+
+Copyright (c) 1999 RidgeRun, LLC.
```
2019

michaelgruner commented on configure.ac:
```diff
@@ -1,8 +1,10 @@
+dnl package name and package version
+dnl NOTE: whenever you change this, also update debian/changelog
+dnl (e.g., run `dch -i`).
+AC_INIT([gst-perf],[0.2.5])
```
We follow a home-grown version [semanthic versioning](https://semver.org), so given `major.minor.micro` we reserve the `micro` for critical bug fixes. This would be a `minor` change, but we don't increase them with every new feature. We typically group a bunch of new features and then do a release where we bump the version.

Having said that, I think that this version change will be done by the engineer making the release (probably me) in a "release" branch.

michaelgruner replied:
I do like the comment though, please keep it.

michaelgruner commented on debian/control:
```diff
@@ -0,0 +1,13 @@
+Source: gst-perf
+Priority: optional
+Maintainer: Melissa Montero <melissa.montero@ridgerun.com>
```
Please use `RidgeRun Developers <support@ridgerun.com>`

michaelgruner commented on debian/changelog:
```diff
@@ -0,0 +1,5 @@
+gst-perf (0.2.5-1) UNRELEASED; urgency=medium
```
See comment about version in the `configure.ac`.

michaelgruner reviewed (changes requested):
Thanks for the contribution @cxw42! Do you want to apply the changes yourself or do you want us to take it from here?

cxw42 commented:
@michaelgruner Thanks for the review!  I am about to force-push the following delta (from 1b33d3b to 2b35a3e):

```diff
diff --git a/README.md b/README.md
index 06a89b9..135cd4e 100644
--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ gst-launch-1.0 -e videotestsrc ! x264enc ! perf ! qtmux print-arm-load=true ! fi
 
 ## Legal
 
-Copyright (c) 1999 RidgeRun, LLC.
+Copyright (c) 2019 RidgeRun, LLC.
 
 Portions copyright (c) 2020 D3 Engineering, LLC.
 
diff --git a/configure.ac b/configure.ac
index 49e236c..28bdd3e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4,7 +4,7 @@ AC_PREREQ([2.53])
 dnl package name and package version
 dnl NOTE: whenever you change this, also update debian/changelog
 dnl (e.g., run `dch -i`).
-AC_INIT([gst-perf],[0.2.5])
+AC_INIT([gst-perf],[0.2.4])
 
 dnl required versions of gstreamer and plugins-base
 GST_REQUIRED=1.0.0
diff --git a/debian/changelog b/debian/changelog
index e285b7b..046ebc2 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,4 +1,4 @@
-gst-perf (0.2.5-1) UNRELEASED; urgency=medium
+gst-perf (0.2.4-1) UNRELEASED; urgency=medium
 
   * Initial .deb release
 
diff --git a/debian/control b/debian/control
index 41bdb16..bb27860 100644
--- a/debian/control
+++ b/debian/control
@@ -1,6 +1,6 @@
 Source: gst-perf
 Priority: optional
-Maintainer: Melissa Montero <melissa.montero@ridgerun.com>
+Maintainer: RidgeRun Developers <support@ridgerun.com>
 Build-Depends: debhelper (>= 8.0.0), devscripts
 Standards-Version: 4.1.0
 Section: utils
```

Commit:
Add .deb packaging support

- Add debian/ files to permit building .deb
- Add .deb-building instructions to README
- Update .gitignore to exclude packaging files

michaelgruner reviewed (changes requested):
Source looks alright! Please rebase to latest `dev-0.3` and I'll merge there.

cxw42 commented:
@michaelgruner it was easier to cherry-pick on top of dev-0.3 since dev-0.3 has diverged a fair bit from master.  I just force-pushed this so now it is on dev-0.3.

Thanks very much for the Hacktoberfest credit! :D

michaelgruner commented:
On the contrary, thanks to you for the contribution! I have eyes on that frame drop branch of yours 👀

cxw42 commented:
@michaelgruner thanks!  The frame-drop branch is ready for preliminary testing --- if you have a chance to give it a try and let me know your experience, I'd be happy to receive bug reports :D .  It's not ready for a PR, though.
